File: 3-ordenacao/CollectionJava/Aluno.java
public class Aluno implements Comparable<Aluno> {
    public int idade;
    public String nome;

    public Aluno(int idade, String nome) {
        this.idade = idade;
        this.nome = nome;
    }

    @Override
     public boolean equals(Object o) {
        if (o instanceof Aluno) {
            Aluno e = (Aluno) o;
            return this.idade == e.idade && this.nome.equals(e.nome);
        }
        return false;
    }
    
    @Override
    public int compareTo(Aluno a) {
        if (this.idade < a.idade) {
            return -1;
        } 
        if (this.idade == a.idade) {
            if (this.nome.compareTo(a.nome) < 0) {
                 return -1;
            }
        }
        return 1;
    }

    @Override
    public String toString() {
        return "Nome: "+ this.nome + ". Idade: " + this.idade; 
    }
}

File: 5-hash/Csharp/Aluno.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Csharp
{
    internal class Aluno : IComparable<Aluno>
    {

        public int Matricula { get; set; }
        public String Nome { get; set; }

        public Aluno(int matricula, String nome)
        {
            this.Matricula = matricula;
            this.Nome = nome;
        }

        public Aluno(int matricula)
        {
            this.Matricula = matricula;
        }

        public override int GetHashCode()
        {
            int hash = 7;
            hash = 41 * hash + this.Matricula;
            return hash;
        }

        public override bool Equals(Object obj)
        {
            if (this == obj)
            {
                return true;
            }
            if (obj == null)
            {
                return false;
            }

            Aluno other = (Aluno)obj;
            if (this.Matricula != other.Matricula)
            {
                return false;
            }
            return true;
        }

        public override String ToString()
        {
            return "Aluno{" + "matricula=" + this.Matricula + ", nome=" + this.Nome + '}';
        }

        public int CompareTo(Aluno a)
        {
            if (this.Matricula == a.Matricula)
            {
                return 0;
            }
            else if (this.Matricula < a.Matricula)
            {
                return -1;
            }
            return 1;
        }
    }
}
